src/gennylib/src/encryption.cpp

```diff
@@ -305,6 +332,55 @@ void FLEEncryptedField::appendEncryptInfo(sub_document subdoc) const {
     }
 }
 
+QueryableEncryptedField::Query::Query(const Node& yaml) {
+    queryType = yaml["queryType"].to<std::string>();
+    contention = yaml["contention"].maybe<int>();
+}
+
+QueryableEncryptedField::QueryableEncryptedField(const Node& yaml) : EncryptedField(yaml) {
+    const auto& queriesNode = yaml["queries"];
+    if (!queriesNode) {
+        return;
+    }
+
+    if (queriesNode.isSequence()) {
+        for (auto& [_, queryNode] : queriesNode) {
+            if (!queryNode.isMap()) {
+                throw InvalidConfigurationException(
+                    "Each value in the 'queries' array must be of map type");
+            }
+            _queries.emplace_back(queryNode);
+        }
+    } else if (queriesNode.isMap()) {
+        _queries.emplace_back(queriesNode);
```

Comment:
There doesn't appear to be a valid test for this.

Reply:
I added a test for `EncryptionContext::generateEncryptedFieldsMapDoc()`, which should cover the valid cases for this Node.